src/Employer/Employer.Web/Controllers/VacancyManageController.cs

```diff
@@ -21,9 +24,12 @@ public VacancyManageController(VacancyManageOrchestrator orchestrator, IEmployer
         }
 
         [HttpGet("", Name = RouteNames.DisplayVacancy_Get)]
-        public async Task<IActionResult> DisplayVacancy([FromRoute]Guid vacancyId)
+        public async Task<IActionResult> DisplayVacancy(VacancyRouteModel vrm)
         {
-            var vacancy = await _client.GetVacancyAsync(vacancyId);
+            var vacancy = await _client.GetVacancyAsync(vrm.VacancyId);
+
+            if (!vacancy.EmployerAccountId.Equals(vrm.EmployerAccountId, StringComparison.OrdinalIgnoreCase))
```

Comment:
Should this not live in the Orchestrator?

Reply:
Hi, the reason why I am using the client in the controller is because I need to inspect the status of the vacancy first before branching out into either calling a method that calls redirection or calling another method on the orchestrator which does inspection of the same vacancy. This was all to avoid calling GetVacancy on the client twice or having a method on the orchestrator which calls the client to return a Vacancy to the controller. I can change this if you prefer?

Reply:
Have refactored this so the orchestrator is doing most of the work and calling the EmployerClient instead of the controller